lib/strings.nix

```diff
@@ -661,6 +661,40 @@ rec {
       name = head (splitString sep filename);
     in assert name != filename; name;
 
+  /* Create a -D<feature>={enabled,disabled} string that can be passed to
+     typical Meson invocations.
+
+    Type: mesonFeature :: string -> bool -> string
+
+     @param feature The feature to be enabled or disabled
+     @param flag The controlling flag that will enable or disable the feature
+
+     Example:
+       mesonFeature docs true
+       => "-Ddocs=enabled"
+       mesonFeature savage false
+       => "-Dsavage=disabled"
+  */
+  mesonFeature = feature: flag:
+    "-D${feature}=${if flag then "enabled" else "disabled"}";
```

Comment:
Can we make this a wrapper of a third function which allows passing custom strings

```
mesonNotSureYet "feature" "auto"
```

Reply:
I like it, but `mesonPleaseINeedAGoodNameForThis`.

Reply:
Well, we have something like

```
-D${feature}=<true|false|enabled|disabled|customString>
```

Maybe `mesonOption feature value` to start with?